client/ayon_deadline/plugins/publish/global/validate_expected_and_rendered_files.py

```diff
@@ -41,34 +42,10 @@ def process(self, instance):
             staging_dir = repre["stagingDir"]
             existing_files = self._get_existing_files(staging_dir)
 
-            if self.allow_user_override:
-                # We always check for user override because the user might have
-                # also overridden the Job frame list to be longer than the
-                # originally submitted frame range
-                # todo: We should first check if Job frame range was overridden
-                #       at all so we don't unnecessarily override anything
-                collection_or_filename = self._get_collection(expected_files)
-                job_expected_files = self._get_job_expected_files(
-                    collection_or_filename, frame_list)
-
-                job_files_diff = job_expected_files.difference(expected_files)
-                if job_files_diff:
-                    self.log.debug(
-                        "Detected difference in expected output files from "
-                        "Deadline job. Assuming an updated frame list by the "
-                        "user. Difference: {}".format(sorted(job_files_diff))
-                    )
-
-                    # Update the representation expected files
-                    self.log.info("Update range from actual job range "
-                                  "to frame list: {}".format(frame_list))
-                    # single item files must be string not list
-                    repre["files"] = (sorted(job_expected_files)
-                                      if len(job_expected_files) > 1 else
-                                      list(job_expected_files)[0])
-
-                    # Update the expected files
-                    expected_files = job_expected_files
+            is_image = f'.{repre["ext"]}' in IMAGE_EXTENSIONS
```

Comment:
As far as I know `representation["ext"]` is not 100% required to be set, it used to be totally valid to have representation data as e.g. `{"name": "exr"}` without specifying `ext`.

Reply:
Is it? I though it is requirements...

Reply:
Hm, made logic bit safer.

Reply:
Actually - I did a search over the codebase and didn't quickly find any place where we're still generating it without `"ext"`. But I'm quite sure it used to be optional - but it doesnt seem to be now anymore...

- It's listed as required here: https://github.com/ynput/ayon-core/blob/9ea9e344a2e147e1681a291795f3352ea71cb5af/client/ayon_core/plugins/publish/integrate.py#L100
- Plus we're directly accessing the `ext` key here: https://github.com/ynput/ayon-core/blob/9ea9e344a2e147e1681a291795f3352ea71cb5af/client/ayon_core/plugins/publish/integrate.py#L542

So ignore me, definitely doesn't seem optional anymore. 👍

Reply:
Sorry about that @kalisp - I'd say, feel free to revert... and then if we face errors somehow we should be fixing the `ext` requirement elsewhere most likely.

Reply:
It was optional, when we started to use it, but that's a long time ago.

Reply:
But there is some issue for representation `as a folder`, so what will be in those?

Reply:
I don't have answers to everything, my guess is that the `"ext"` should be filled to `None`, which should not break logic here.

Reply:
> But there is some issue for representation `as a folder`, so what will be in those?

Those are still required to have `ext` **plus** that is currently NOT implemented so can be disregarded for this PR I'd say.